### ISSUE TITLE
Nomic, git and tinder.

### ISSUE THREAD

**Issue by mm0hgw** [TheDemocracyFoundation/epitome#45]
http://legacy.earlham.edu/~peters/writing/nomic.htm

Nomic forms a unit test for your proposal. If the project can run a game of Nomic, it can run an arbitrary legislature.

git forms a storage mechanism for a body of legislature arbitrarily modified by amendments.

tinder manages to crowdsource a large number of proposals into a large number of reviews.

**Comment by mm0hgw** [comment]
User roles : elector, legislator, executive, adjudicate.

Executives propose amendments to the legislature.
Legislators review, rate and propose amendments to the amendment.
Electors vote on amendments.
Adjudicators review complaints where there is dispute about what the legislature means, based upon what it says.

The collective will likely require a chief executive to negotiate contracts with external parties on behalf of the collective, and a treasurer to manage the assets of the collective.

**Comment by pbout** [comment]
Thank you for your input. At the current stage, we are trying to reach a bare minimum functionality state. During this time user roles like an administrator are required as some actions will be performed manually. Our goal is to have users not be restricted by other users with elevated privileges, but by user-defined rules. Our concept could potentially run a nomic game, as in that scenario, users begin with a predefined set of rules which however they can modify while “playing”. In our solution, we have the concept of cycles which could be considered turns from the perspective of a nomic player (albeit with several failsafe mechanisms to prevent deliberate or accidental abuse). Similarly to nomic, the mechanism to prevent “a brief wave of fanaticism from undoing decades or centuries of refined structure” (immutable law), is both vote and time restricted.  An “immutable law” would not be modified for a predefined period of time and would require a supermajority vote.

**Comment by mm0hgw** [comment]
The reason a judiciary is a fundamental requirement for a legislature is that you can't rely on people to know what words mean.

If you haven't achieved a functional judiciary, you haven't achieved bare minimum functionality. Sure you can pass legislation, but you've no mechansism to know what the words that construct the rules mean.

You haven't even decided between a civil law system and a common law system. In civil law, the duty of the judge is to rule acording to the spirit of the legislature. In common law, the duty of the judge is to rule according to the letter of the legislature. There's a contraversial comma in the 2nd amendment to the US constitution. Under the French system the legal argument would not be about the text of the 2nd amendment, but about the intent of those who proposed, reviewed and passed the amendment.

git stores computer code. Computer code requires a unique form and an explicit meaning to each word. You can't write a complier without these features.

Unlike computer code, human langauge has neither unique forms, nor explicit meanings. Thus for the law to have meaning fundamentally requires a judiciary.

The only way to avoid this is to write law in language without the potential to be ambiguous. Until this is done, there will be a need for lawyers wherever there is law.

**Comment by mm0hgw** [comment]
Given there is a need for a judiciary, there's a need to crowdsource judicial services. I envisaged the following system.

Anyone can vote, or make complaint. Both are optional.

Voting earns engagement points.

After sufficient engagement, you gain the ability to rate potential legislative amendments. Only the highest rated amendments are actually put to the electorate.

After sufficient engagement, you gain the ability to propose new amendments and amendments to amendments.

After sufficient engagement, you gain the ability to provide judicial services to the community.

Exactly what qualifies as 'sufficient' depends upon the size of the lexicon of extant legislature.

**Comment by pbout** [comment]
"The reason a judiciary is a fundamental requirement for a legislature is that you can't rely on people to know what words mean."
This in itself is contradictory to your initial proposition of using nomic as a unit test. It would be impossible to even begin the game if one was not able to understand, or lacked the means to gain an understanding of the meaning of a rule.

We have already considered a similar point driven system where the interaction of a user would grant him elevate privileges but we are generally against that. Reasons being, receiving points for voting would encourage careless voting. Large amounts of points would not be proof of experience of the voter. 

We have addressed several of your concerns in our whitepaper which also contains a concentrated summary of how our proposed system would work. http://democracy.foundation/epitome/

**Comment by mm0hgw** [comment]
I think you misunderstood.

It's not that it is somehow impossible to understand all words.

It is that it is impossible to guarantee every member of the electorate understands every single word used in the legislature in exactly the same way. Dictionaries have existed for centuries, yet the meaning of words in common usage is still in constant flux, and lawyers have not been rendered obsolete.

It is perfectly possible to have a game of Nomic where two or more players disagree on the correct interpretation of a rule. It's probably vanishingly rare to have a game where this does not happen. 

It is also possible to have a game of Nomic where some players don't bother to learn the rules, or are not capable of understanding the rules.

Typically in Western democracies, about a third of the electorate don't bother voting even when sovereign power is on the line, expressing little interest in the rules.

If you're going to pretend every member of the electorate is as politically engaged as you think they should be, you're going to be disappointed.

Australia levies fines on those who passively abstain. What can your system do to encourage sufficient participation?

**Comment by mm0hgw** [comment]
You may have seen a box labelled 'I have read and understood the terms and conditions.'

These are a legal nonsense, as while you certainly have a good idea about whether you've read a text, to have a good idea about whether you've understood a text requires independent (of the entity that wants you to agree) legal advice.

An agreement made without access to independent legal advice can be claimed to be void on the grounds it was made under duress.

Fundamentally the problem is, if you don't understand some part of the legislature, but *think* you do, how are you supposed to know you don't without independent legal advice?

**Comment by mm0hgw** [comment]
https://en.m.wikipedia.org/wiki/United_Nations_Convention_on_Contracts_for_the_International_Sale_of_Goods

tl;dr Contract negotiation is a sequence of 'pull requests,' with both 'offer' and 'counter-offer' constituting a commit of diffs from the status quo.

Each 'offer' requires:
- threaded discussion for all users w/ karma
- accept/reject poll for legislative users
- facility to propose amendments for executive users
- recourse to adjudicative users for any relevant query or complaint.

If legislative, executive and adjudicative roles are restricted to volunteers, then one knows when a majority of each volunteer group has been achieved. This is useful in assessing when a majority of the voluntary legislative group has accepted a proposal should be put to the electorate.

If, instead, one conscripts the entire electorate to act as legislators whether they want to or not, one should expect the quality of the product to suffer relative to the same group with voluntary legislators.

Shoddy workmanship is the price of conscripting everyone to avoid legislator being a 'priviliged' position.

**Comment by MavropaliasG** [comment]
@mm0hgw your ideas are great. At present, we are trying to create a system of deliberation and consensus-decision making which will be completely independent of moderators or representatives. That by itself is a very challenging task. Initially we are focusing on small groups of people: organisations, universities, NGOs etc. with no intention to focus on becoming a national governance platform. However later on, once we have established a good operating deliberation system, we will start looking into developing features that can be applied in communities and societies and that will render the development of laws possible. But we will still try to create a system that will have complete independence from representatives, intermediates, or moderators.

**Comment by MavropaliasG** [comment]
One method that we have considered is to create a syntax for legal language. Similar to how R has predefined commands and expects specific input to them (variables, arguments, values, objects etc) legal language could also be developed with a specific syntax and specific required inputs. That way the deliberation system  with the diff function (our whitepaper, figure 2) could show amendments done by more experienced users similar to programming code, and improve the writing of an idea

**Comment by mm0hgw** [comment]
Whitepaper 2.2

Four elements are listed. None of them is people. To ensure each user only votes once, users must be uniquely identifiable.

Basically you need industrial-grade user tracking, like a Kerberos/LDAP based SSO system.

Whitepaper 3

Every member of the electorate is conscripted to cycles, when observably, ~1/3rd of the UK/US electorate doesn't make a decision even when sovereign power is on the line.

MavropaliasG asked about the relevance of Nomic. Both Nomic and international law on contract negotiation are sets of rules for generating sets of rules. The UN CISG model is superior in that it is Turing Complete and scaleable.

Each proposal has only 3 possible responses: accept, reject and counter-proposal.

The counter proposals effectively form different branches of a git repo. When a legislator accepts a proposal branch, they implicitly reject all others. Dynamically, one can display the remaining branches capable of achieving sufficient accepts to achieve a majority of legislators. 
If we define 'legislators' as those who've voted on the legislative system in this or the previous legislative session, then we've a decent estimate of political engagement with the system. Similarly, 'executives' are the formal authors of proposals, and adjudicates volunteer to handle queries about and complaints under the rules.


After the legislative period, an elective period occurs where those executives who've had a proposal accepted get to attempt to evangelise their proposal to the electorate. Formally the Chief Adjudicate takes upon the role of opposition, or 'assessing the threat the proposals pose to the rules.'

After the elective period, the electorate votes on the proposals accepted by the legislative.

Those proposals passed are, typically, to be enacted by the proposer.

Adjudicates are required because users reserve the right to ignorance. A mod would typically flag an opinion, query or thread for adjudicative advice.

As well as queries, adjudicative users must also handle complaints, where a plaintiff seeks reparations from a defendant under the rules.

Typically in a small business deal, one agrees to settle disputes by arbitration, which is faster, cheaper, private, and doesn't set any nasty precedents.

- both defendant and plaintiff agree to abide by the results of arbitration.
- both defendant and plaintiff select an arbiter.
- the two arbiters select a third arbiter
- the three arbiters assess if there is sufficient substantiative evidence to uphold the plaintiff's complaint, and come to a decision about liability.
- if both defendant and plaintiff assent, the proceedings may then be made public.
- The Chief Adjucate is formally required to defend the collective, so can't act as arbiter where the collective is defendant.

Duplicates are recognised by acceptance by a simple majority of adjudicates. This merges the support for the two branches under the one with the earlier timestamp.

**Comment by mm0hgw** [comment]
I've a semi-deranged idea for a model there. I like the idea of everyone being able to see the current state of declared votes, and redeclare their vote at any time. The votes would only complete when majority assent was achieved.

**Comment by mm0hgw** [comment]
Also, you've an entire section on 'evidence.'

According to President Trump:

- No substantiative evidence for man-made climate change has ever been collected.
- Artificial climate change is not important.
- No-one has been affected.
- Artificial climate change didn't happen in a manner or for a reason.
- Artificial climate change hasn't occured anywhere.
- Artificial climate change hasn't occured any time.
- The cost is lots of pointless, unsubstantiated debate on an issue that isn't happening.
- The bickering will last as long as fake news is a thing.

This is conjecture rather than opinion, but Trump doesn't care because he can't tell the difference between his subjective opinion and objective evidence. After all, being rich, he must know how the world actually works, so why wouldn't his opinion be as weighty as his wallet?

**Comment by MavropaliasG** [comment]
@mm0hgw great ideas as usual. However I do think that you are focusing too much on the national application of this software again. Yes that is the end goal, but we have a lot of way until then, and we are focusing on smaller applications now. After all, all the direct democracy projects that failed ( http://democracy.foundation/similar-projects/ ) aimed to replace politicians straight away and convert countries to direct democracy, and that is why they failed. People need to get used to voting in their everyday lives, so that is why we are focusing on organizations and small groups first. When the time comes and the software is widely used, then we will start implementing some new features for legislation development and integration with an already existing system of representative democracy (probably in small villages or towns).

But when we reach that stage, all the systems described in our whitepaper will have evolved a lot. I still believe thought that we will be able to find methods that no “authority” will be required anywhere in the process, from the identification of an issue in a society, to its final solution or legislation drafting.

Just to clarify though, even with our present design, no one needs to “pick” which proposals are final. If you read sections 5.4 and 6, the proposals with the most support are simultaneously introduced as referendum of ranked voting. So maybe in the future, “legislators” or people who can write legal language can enter the process once a proposal is gaining traction to start converting it to legal language, it might sound as a difficult task, but there are thousands of people with that ability and crowdourcing it will produce a much better result than it would be by a few members.

However you keep proposing processes with intermediates. We will keep trying to think of methods that no intermediates, authority or “experts” are used, as having them would be no different from our current, representative system in the long term.

A Trump would have been lost in the deliberation map, as his voice would have the same power as anyone else. If you made the deliberation anonymous, that would extinguish any prejudice or “power” - everyone’s voice would have the same power.

**Comment by mm0hgw** [comment]
4chan doesn't have moderators, intermediates, authority or experts. Have you tried having this conversation on 4chan?

**Comment by MavropaliasG** [comment]
The thing with 4chan is that there is no moderation whatsoever, whereas the deliberation section we're trying to develop would be moderated by the users themselves. The "33-60" rule (section 4.4) would dictate that a representative portion of the users consider a submission as inappropriate, and then remove it. So the mass-moderation of the users would remove any single point of abuse and spread the moderation equally among everyone.

**Comment by mm0hgw** [comment]
So of you've a voting bloc that can hit 33-60, you can utterly silence any opposition as 'spam?'

**Comment by MavropaliasG** [comment]
@mm0hgw it would have to reach 60% agreement among 33% of the current active population. In a cycle of only 10 people, that would be easy yes, but if you applied that to a town or a country it would be impossible to have an organized action to silence opposition since users are assigned randomly to cycles. So your "friends" would be in other cycles. Moreover a successfully reported node is simply hidden, with the ability to view and also trigger a reverse vote, in case something like that did manage to happen (whitepaper 4.4). So if this system is applied to 10 or 10 million, as the cycle levels divide the population to manageable amounts, but still keep a random sample in each cycle, it would provide a nice representative sample of the population in each cycle, and in case power groups did manage to form, reversal of their actions is possible.

**Comment by mm0hgw** [comment]
So in a group of 10, 33% of the active population is 10/3, and 60% of that is 2. So 2/10 = 20% people can shut down debate on anything?

**Comment by mm0hgw** [comment]
If you're randomly sampling your cycles from a population that is ~1/4 Left, ~1/4 Right and ~1/3 Abstainer,  you don't even need bipartisan support to prevent, say, abortion from being discussed?

**Comment by MavropaliasG** [comment]
If such a small group of 10 people would use a software like epitome that is meant for larger populations, then yes 2 people would be able to silence a node. However the other 8 would be able to unhide and reverse the decision.

**Comment by mm0hgw** [comment]
What is the target group size?

My proposal was for a target group from 2 (small business deal with external arbitration) or more.

Fundementally, you *must* have people who can represent the collective. If the collective does not empower anyone to represent it, it cannot form contracts with anyone. If it cannot form contracts with anyone it cannot apply any sanction for breaking the resolutions of the collective, nor reward for following them. If there is no consequence to breaking a resolution of the collective, what do they mean?

The smallest self-contained collective is 3. If 1 has a formal complaint about 2, then 3 could volunteer to act as an internal arbiter. If 3 refuses, then 1 must look externally for arbitration.

You seem to have misunderstood some definitions. The 'legislators' are not those who *can* vote on legislative proposals, that group is the electorate. 'Legislators' are that subset of the electorate that *do* habitually vote on legislative proposals.

Similarly, 'executives' are not those who can make proposals. Those outside the electorate must be able to make proposals for negotiation. 'Executives' are those members of the electorate who *do* make (and typically also enact) proposals which are ratified by the electorate.

No-one is excluded from any group, the groups are defined by the individual actions of the electorate.

**Comment by mm0hgw** [comment]
The need for advanced user identification is for audit purposes. It applies even to the case of 2 people negotiating a small business deal. The system is required to command confidence that each proposal and each comment in the discussion thread is genuinely made by the reported user. If there's some security flaw, and user id is trivial to spoof, this cripples any confidence that can be had in any resolution generated by the system.

**Comment by mm0hgw** [comment]
https://djangopackages.org/grids/g/ldap/

LDAP is optimised for read-heavy operation, and the write operations in iterative proposal development and dynamically counted democratic ballots are limited.

**Comment by mm0hgw** [comment]
https://pypi.python.org/pypi/django-kerberos

Kerberos is already available for django.

**Comment by mm0hgw** [comment]
Re: A Trump would have been lost in the deliberation map, as his voice would have the same power as anyone else. If you made the deliberation anonymous, that would extinguish any prejudice or “power” - everyone’s voice would have the same power.

http://www.independent.co.uk/news/world/americas/us-politics/donald-trump-tweet-fake-polls-news-lowest-approval-rating-historic-latest-a8127316.html

Trump has 'historically low' approval ratings of .... 35%, so he'd have the 20% support required to achieve 33-60 in the majority of cycles.

**Comment by MavropaliasG** [comment]
Thank you @mm0hgw , your ideas are very interesting. There are many projects that include people who will represent others or moderate the program. Epitome will not be one of them. That is the challenge we’ve set to achieve. We are far from reaching it, but one day we will be able to create system completely independent of representatives, with all the member being the moderating backbone. A team of 3 or even 10 members may need to reach consensus, but Epitome is not the tool they should use. Each tool has a specific use, and Epitome will be created for large groups, small groups can reach consensus even in a chatroom.

I do understand your separation of powers in your model, but we simply do not want to separate powers. We want to create a method that the people will report issues and generate solutions. If the solutions are in the form of legislation, so be it, as denoted by our whitepaper and the braces {} that we put around the numbers and features, our platform will be very customizable, so that in the distant future if a town would decide to make it its primary legislative instrument, then they could customize it however they want, and for example could have the whole process from the deliberation to the referendum for the selection of the most preferred solution from the people, be the method that the government would find which proposal is the most popular and take that to build its next bill. So it won’t have to be binding that the proposal that won in the final referendum will be the law, but it will be the idea that the subsequent law will be based on. We want to make Epitome flexible and its applications many.

Thank you so much for the links of the django plugins. We will definitely implement them. We are still trying to develop basic usability, heavy security is not our main focus – for now.

Finally about Trump (funny how any conversation nowadays sooner or later includes Trump). A world where this system would be the primarily tool of governance will have no influence of Trumps, else they would not allow the establishment of this tool. You seem to assume that the USA is a good model for direct democracy. That can’t be further from the truth. USA will be one of the last countries to transition to direct democracy, with the first ones I presume will be Iceland and Switzerland.

The idea of decentralization and direct democracy is the future of humanity, yet no one seems to build anything more than simple voting tools with disappointing design. What we want to create is a new method of communication, from which other tools, from other team might sprout and cause a pivotal moment in human societies.

**Comment by mm0hgw** [comment]
You've defined a meaningless task. There must be some consequence to failing to follow rules for the rules to have meaning. As I said, if you refuse to empower individuals to sign agreements, contracts or treaties on behalf of the collective, then the collective cannot have any agreements, contracts or treaties.

Civil liability cannot be assigned to a person or group without the agreement of that entity. That entity cannot agree to accept any consequence of breaking the rules if the collective declines to provide anyone to agree on behalf of the collective.

**Comment by mm0hgw** [comment]
I don't think *I* understand the separation of powers in my model. I'm not sure how you think the 'powers' are separated. I'm not sure what you think the 'powers' are. I didn't define 'powers' I defined 'responsibilities.'

**Comment by mm0hgw** [comment]
Also, another time you must have a single designate is if an epitome collective is taken to court. If your collective refuses to supply the court with a representative, the court can't be expected to believe the group is in any way organised.

**Comment by mm0hgw** [comment]
"The idea of decentralization and direct democracy is the future of humanity"

@MavropaliasG it's clearly what you think the future should be. That doesn't make it likely or even possible. With some pragmatic compromise we could be on to something.

I've experienced direct democracy. It needs threaded conversation.

Democracy isn't how we run things in the west. We run things by investing money in ordinary shareholdings.

**Comment by mm0hgw** [comment]
https://en.m.wikipedia.org/wiki/Request_for_proposal
https://en.m.wikipedia.org/wiki/Invitation_for_bid

If you've a design and need to knock out a few thousand units, there are more effective things to do than mess about with direct democracy.

The compromise I envisage is to use git to rapid-prototype proposals, and LDAP to store voting information on the process and kerberos to identify everyone.

If one has to achieve majority assent among the legislators, then one will do a lot of rapid prototyping before bothering the non legislative electorate. For a sane electorate, the proposals they're bothered with need to be few, and well-developed.

**Comment by MavropaliasG** [comment]
I may start becoming repetitive, but you keep raising the same arguments. As I said, our focus and design for now is not for legislation development, but more of a communication tool. There can’t be failure to follow the rules as there are no rules.

To describe what our initial design will do as simple as possible, is that it will allow groups of people in their community (and when I say community I do not mean an actual physical community of people, but any community such as an NGO or an open source community etc) to report issues/problems, and through a deliberation method that we have design, to produce a referendum with the proposals that have the greatest popular support. We don’t want to overwhelm ourselves by trying to create all the features required to have a fully autonomous direct democracy national governance system, that would be insane and pointless. We want to create a fist version of the platform that will be able to successfully produce solutions to reported problems and build from there on. We have made no designs or tried to apply the software that we’re building on real societies and try to integrate it with the actual legal system. That will not be our focus for at least the next 5 – 8 years. We first want to get a lot of organizations and online communities use this new method and practice participation in the governance of their groups. We want to get people used to the idea of participation and voting, and to hone their critical and governance skills.

The arguments your raised about actual governance or even a digital court/trial are very good and we have already thought of future expansion in those areas but we are certainly not trying to design them as features yet.

I like your ideas but I think if we implement our deliberation system with the interactive map and all those visual indications, it will be superior to what git can do for both amateur and advanced users. 

Take a look at our planned “journey”  http://democracy.foundation/roadmap/

**Comment by mm0hgw** [comment]
No rules? But agreements? I think you misunderstand what agreements are.

Where actors A and B agree that A shall provide B with 20,000 eggs for 20,000 dollars, on the first of every month, then this forms a rule that binds both parties. If B fails to provide the money, or A fails to provide the eggs, on the appointed day, the (agreement/deal/rule) has been broken and the injured party has recourse to claim with the court.

If the parties have no recourse in the event that agreement is broken by the other party, they have no reason to keep to the agreement. An agreement that isn't a rule isn't functional. The whole point of being an agreement is to be a rule that binds the agreeing parties.

**Comment by mm0hgw** [comment]
A receipt demonstrates the bare minimum legally acceptable agreement. It lists the items agreed and the date they were agreed. 

Without a date of agreement, it is not possible to prove the agreement applied at any time, so it is not possible to prove the rule was binding upon the defendant at the time the plaintiff alledges the defendant broke the agreement.

There are three things you just can't legally be fuzzy on with agreements more complex than a simple transaction:
- the exact text that was agreed.
- the identities of the agreeing parties
- the date of agreement

The need to be exact on these doesn't suddenly arise when trying to run a country. Companies, NGOs, GOs and private individuals all have a requirement to be exact about the agreements they make. 

"I like your ideas "
The UN CISG 1980 wasn't exactly my idea. 

" I think if we implement our deliberation system with the interactive map and all those visual indications, it will be superior to what git can do for both amateur and advanced users."

I think I haven't seen anything to suggest your deliberation system will improve democracy. Armies are not commanded by democracy, because democracy takes too long to make decisions.

git can stich together proposal trees, cryptographically authenticate when and by whom each line of the proposal was submitted, and make anonymous edits impossible.

https://github.com/MavropaliasG/R-Scripts/blame/master/Scripts/swapHalves.R
This is stitched together from different sources. When and by whom is recorded.

When your electronic records of the acceptance of the proposal have the hash of the proposal commit (as proof of reading) and timestamps later than the proposal they were accepting, you have a record of an agreement.

If you think you can do better than git, you underestimate how much cryptographic authentication you'll have to reinvent to do so.

**Comment by mm0hgw** [comment]
https://en.m.wikipedia.org/wiki/Online_dispute_resolution

Digital court already exists.

**Comment by MavropaliasG** [comment]
Thank you for your comments @mm0hgw  . Epitome will not be a tool to create contracts and enforce them between two parties. It will be a tool to allow the representation-less consensus creation within a community. This consensus, will not be automatically enforceable. I think the contract creation and enforcement is best suited for blockchain applications. Maybe we can create a feature in the future for contract creation to find a way to enforce referendum decisions, but this will probably be in our next platform, which will be blockchain (or IOTA) based. However our focus for now is to create a new method (incomplete at first as is everything new) for mass deliberation and consensus creation. I cannot stress this enough.

“I think I haven't seen anything to suggest your deliberation system will improve democracy. Armies are not commanded by democracy, because democracy takes too long to make decisions.”

Each system has different capabilities which render it a good choice in some situations and bad in others. Certainly an army or for example services that deal with emergencies could never be operated by direct democracy. However the collective of a group could select people to put in the places where there is a need for rapid decisions, and when decisions are not needed to be taken immediately, that person could create polls within our platform to get the group’s opinion.

I have seen that git contributions can be visualized in a map close to what we have designed, for example with http://gource.io/ . When the time comes and we would like to start working on the deliberation section, we might reconsider git. However I think that git is fundamentally build for another purpose, and trying to customize it and keep hacking it to fit our purpose, will be more difficult in the long term than simply creating something simple and the building it further on. But as I said, we will study the different options when the time comes. The “when and by whom” as you said is quite useful as well as the good cryptographic authentication that git has. Thank you for this.

I would love it if you kept contributing your ideas in the future as well, your suggestions have provoked quite some thought and discussion.

**Comment by mm0hgw** [comment]
"It will be a tool to allow the representation-less consensus creation within a community."

https://en.oxforddictionaries.com/definition/consensus

But you're redefining consensus so it *doesn't* involve agreement?

epitome is a tool for people who agree with you that consensus is not agreement, to create consensuses that aren't agreements? That's a short list.

It's not directly your choice which agreements are legally enforceable and which are not. Any agreement where the identity of the parties, text of the agreement and date of commencement are reliably recorded is legally enforceable.

To ensure the consensuses reached by epitome are definitely not legally enforceable, you must obscure/destroy so not even the users know at least one of:
- the identity of those in the consensus
- the text, subject of the consensus
- the date of the consensus

Which shouldn't be a problem as you've specifically defined epitome is not a tool for making decisions that matter to anyone.

How many meaningless decisions do you need the help of a large group of peers to make?

**Comment by mm0hgw** [comment]
"However the collective of a group could select people to put in the places where there is a need for rapid decisions,"

If you put someone in place to make decisions on your behalf, that person is a *de facto* representative. To have representation-less consensus creation, you must have no representatives.

**Comment by mm0hgw** [comment]
" I think that git is fundamentally build for another purpose, "

git is designed to store text under development, and is optimised for iterative development of that text, while preserving the integrity of historic iterations, and maintaining a small footprint on both hard drive and in wire transit.

If you and your peers are trying to achieve consensus through iterative proposal development using written communication, git is potentially very useful.

If you're trying to achieve consensus through performance art, git is not as potentially useful.

Ooooh, that's good. If it's performance art, it's *expected* that nobody really knows what it means, and thus a consensus generated from performance art can't be legally binding.

**Comment by MavropaliasG** [comment]
Hey @mm0hgw thanks a lot for your comments.

Epitome will have no representatives or intermediates. The whole process from the identification of an issue to the, deliberation, to the crowdsourcing of proposals to the transition of the most prominent proposals to referendums and their voting will be without any moderator, administrators, representatives or intermediates needing to have any sort of input for the process to complete (of course there will be a need for a sysadmin to make sure the server that the platform is hosted is running).

When I was referring to a representative, I was only giving you an idea to the problem you brought up about an immediate action needing to be taken in a nation. Something that we are not focusing now and certainly not in the immediate future. I just wanted to give you a suggestion how a direct democracy system would work in a nation in general. Epitome’s focus is not on national governance at the moment and will certainly do not involve representatives.

BUT, if a corporation would like to use it as a tool, it WILL have the capability to allow only admins to post polls in. As I said we want to create a tool that is very customizable and flexible so, admin-only polls or issues will be also an option. We don’t want to restrict any community by enforcing our opinions, what we want to do is provide something new.

We define consensus in [this way](https://en.wikipedia.org/wiki/Consensus_decision-making). Through the deliberation section, anyone will be able to create “forks” in existing proposal “nodes” and people will be able to shift their support from one node to the next. Users will be able to create “feedback” nodes on existing nodes that raise issues or provide opinions. The most supported nodes are promoted to higher cycle “levels” and start anew the development and collaboration. After the whole process repeats in each level, the most supported nodes are converted to referendums. This is the best method we have thought of creating consensus among potentially thousand users. Live chatting is impossible to follow, but the creation of “forks” in proposals with text amendments being visible and the ability of users to support only limited nodes, is a method that will allow the general user agreement to select the most “fitting” solutions to the best interest of the whole community.

I’m not sure I understand what you said about the generated consensus not being legally enforceable. I do not think you can enforce something if you are not having a person sign it. So far, the best think you can achieve by the successful acceptance of referendum in a community, is an indication of the general will of the community. Nothing is enforceable. This only can be done by contracts such as in the Ethereum blockchain. Correct me if I’m wrong.

When we have a very good and solid software, we will look into developing additional extensions for it. All the ideas you gave could one day be a very good starting point for the development of a new extension.

For git, it seems like a good solution but we will have to look more into it when we decide to being developing the deliberation section.

**Comment by mm0hgw** [comment]
The wiki page on consensus decision making says:

"Consensus decision-making is thus concerned with the process of deliberating and finalizing a decision, and the social, economic, *legal*, environmental and *political* effects of applying this process."

But here you're consistently uninterested in the legal aspects of your decision making tool and would like to pretend it is up to the group how legally credible the evidence produced by the system is, when this just isn't so.

I'm in Scotland. Here, verbal contracts are legally enforceable, so it would be very difficult to argue contracts negotiated by character stream (and are therefore definitely written) were somehow unenforceable.

The *point* of having advanced user authentication is to ensure that absolutely everything in the system is signed. You can make a legally enforceable agreement with amazon without a signature, because they believe you are you, because you know your password, address and credit card details.

If the identities of the agreeing parties, text of the agreement and agreement time are all recorded, then a legal agreement has taken place. Signing a piece of paper is only necessary if you take no other action to identify yourself or signify agreement.

The Debian project seems a little larger than 'a general consensus of the community' since they have Ubuntu downstream.

**Comment by mm0hgw** [comment]
Of course, (IIRC) the Debian project runs on a Schulze method of electing representatives, secured by PKI key. All debian devs have a PKI key.

**Comment by MavropaliasG** [comment]
I can see this conversation seems to have more to it. However it will be best discussed in our discussion group and not as a software issue in github. We have a discussion group that will be of interest to other users as a future reference. In fact, I regret I didn't do that earlier as all this conversation will be lost in the closed issues of github. Please open a new topic here so it will remain in the website and future users can join http://democracy.foundation/discussion-group/
